tensorflow/compiler/aot/codegen.cc

```diff
@@ -229,8 +230,10 @@ Status GenResultMethods(const tf2xla::Config& config,
   int readonly_variables = absl::c_count_if(
       config.variable(),
       [](const tf2xla::Variable& var) { return var.readonly(); });
-  if (config.fetch_size() + config.variable_size() - readonly_variables !=
-      num_results) {
+  const int num_results_int = num_results;
+  const int num_results_comparision_sum = config.fetch_size()
+                                          + config.variable_size() - readonly_variables;
```

Comment:
`comparison`

Also, better name would be `actual_num_results`.

Also, we only need the type of the extracted expression, we can have it match `num_results` and no longer need `num_results_int`